### PR TITLE
Update/fix host-guest example

### PR DESCRIPTION
**Pull request by davidlmobley**
This begins to resolve issues raised in https://github.com/openforcefield/openforcefield/pull/56#issuecomment-334301350 and #66 

To do list:
- [x] Add more complete description of prerequisites for host-guest example
- [x] Fix CB7 case uncovered by @nhenriksen where bond connectivity was broken in intermediate PDB file (bonding was getting perturbed by solvation by `PDBFixer`)
- [x] Work through Slochower issues in #66 and see if anything else still needs fixing or better documenting
- [x] Add documentation of why we need chemical identity of molecules as a starting point

### CODE REVIEW

**Comment by davidlmobley** [comment]
@bannanc - can you review the new markdown material when you have time?

@nhenriksen - if you have time it'd be great for you to check the host-guest example again, though I understand if you don't.

File: Chemical-starting-points.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+Specifically, to apply SMIRNOFF to a system, you must either:
+1. Provide molecules (currently `OEMol` objects, but an RDKit version is in development) corresponding to the components of your system, or
+2. Provide an OpenMM `Topology` which includes bond orders and thus can be converted to molecules corresponding to the components of your system
+
+Without this information, identifying the chemical identity of the molecules comprising your system becomes a matter of guesswork rather than certainty.
+Different molecules can have the same chemical graph but differ in bond order and formal charge, or different resonance structures may be treated rather differently by some force fields (e.g. `c1cc(ccc1c2cc[nH+]cc2)[O-]` vs `C1=CC(C=CC1=C2C=CNC=C2)=O`, where the central bond is rotatable in one resonance structure but not in the other) even though they have identical formal charge and connectivity (chemical graph).
+A force field which uses the chemical identity of molecules to assign parameters needs to know what those molecules are, first.
```
Comment (bannanc) [inline Chemical-starting-points.md]: I don't understand what you're trying to say here.

File: Chemical-starting-points.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+## What about starting from a PDB file?
+
+PDB files do not in general provide the chemical identity of small molecules contained therein, and thus do not provide suitable starting points for applying SMIROFF to small molecules.
+This is especially problematic for PDB files from X-ray crystallography which typically do not include proteins, making the problem even worse.
+For our purposes here, however, we assume you begin with the coordinates of all atoms present and the full topology of your system.
+
+Given a PDB file of a hypothetical biomolecular system of interest containing a small molecule, there are several routes available to you for treating the small molecule present:
+- Use a cheminformatics toolkit (see above) to infer bond orders
+- Identify your ligand from a database; e.g. if it is in the Protein Data Bank (PDB), it will be present in the [Ligand Expo](http://ligand-expo.rcsb.org) meaning that it has a database entry and code you can use to look up its putative chemical identity
+- Identify your ligand from the literature or your collaborators: If the above fail, you will have to find out what ligand is supposed to be in the structure you are working with.
```
Comment (bannanc) [inline Chemical-starting-points.md]: The colon after collaborators here seems off. shouldn't the "If the above... " sentence be on a new line outside the list?

File: README.md [hunk 3]
```diff
@@ -69,6 +69,12 @@ system = forcefield.createSystem(topology, mols)
 See `examples/SMIRNOFF_simulation/` for a complete example of how SMIRNOFF can be used for small molecule vacuum simulations, and `examples/mixedFF_structure` for how to set up a system which uses an AMBER forcefield (in this case, AMBER99SB-ILDN) for a protein in combination with SMIRNOFF for a small molecules. Via ParmEd, this can be translated into GROMACS, AMBER, or CHARMM formats for use elsewhere (and additional formats via InterMol).
 **For an especially complete worked example, see complete setup of a host-guest simulation in water with SMIRNOFF**, including docking, 2D and 3D visualization, etc., in `examples/host_guest_simulation/smirnoff_host_guest.ipynb` (Jupyter notebook).
 
+### FAQs:
+See `Chemical-starting-points.md` for answers to a variety of common problems, such as:
```
Comment (bannanc) [inline README.md]: Can you add a link to the file so people can click on it? Or do we have to add that in a separate PR when that file is in the repo?

File: Chemical-starting-points.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# Appropriate starting points for applying a SMIRNOFF
+
+SMIRNOFF force fields use direct chemical perception meaning that, unlike many molecular mechanics (MM) force fields, they apply parameters based on substructure searches acting directly on molecules.
+This creates unique opportunities and allows them to encode a great deal of chemistry quite simply, but it also means that the *starting point* for parameter assignment must be well-defined chemically, giving not just the elements and connectivity for all of the atoms of all of the components of your system, but also providing the formal charges and bond orders.
+
+Specifically, to apply SMIRNOFF to a system, you must either:
+1. Provide molecules (currently `OEMol` objects, but an RDKit version is in development) corresponding to the components of your system, or
+2. Provide an OpenMM `Topology` which includes bond orders and thus can be converted to molecules corresponding to the components of your system
+
+Without this information, identifying the chemical identity of the molecules comprising your system becomes a matter of guesswork rather than certainty.
```
Comment (bannanc) [inline Chemical-starting-points.md]: This sentence seems a bit weird, I think its that I didn't know for sure what you meant by chemical identity. Maybe something like:
Without this information, our direct chemical perception cannot be applied to your molecule as it requires the chemical identity of the molecules in your system that is bond order and formal charge as well as atoms and connectivity.

**Comment by bannanc** [comment]
@davidlmobley (incase you didn't get notified when I submitted the review)

**Comment by bannanc** [comment]
Also, I didn't look very closely at the notebook changes as they're a little difficult to read in this format, let me know if you want me to open the notebook and read through it.

File: Chemical-starting-points.md (hunk 1 above)
Comment (davidlmobley) [reply Chemical-starting-points.md]: Can you be more specific? I'm trying to say that if you don't have bond order information and formal charges, you don't actually know what molecule you are talking about in some cases, and in other cases, you don't know what resonance structure you are trying to set up, etc. And you can't assign parameters until you've specified that.

Which part seems most confusing?

File: Chemical-starting-points.md (hunk 2 above)
Comment (davidlmobley) [reply Chemical-starting-points.md]: No, the last bullet is intended to be a single thought -- if you can't use a database or a toolkit, you have to work with someone/the literature to figure out what molecule they meant. Maybe I should rephrase a bit, suggestions?

File: README.md (hunk 3 above)
Comment (davidlmobley) [reply README.md]: Can add, yes.

**Comment by davidlmobley** [comment]
@bannanc - thanks for feedback. The notebook is for other folks to review (and has mostly been reviewed).

File: Chemical-starting-points.md (hunk 4 above)
Comment (davidlmobley) [reply Chemical-starting-points.md]: Got it, thanks.

File: Chemical-starting-points.md (hunk 2 above)
Comment (bannanc) [reply Chemical-starting-points.md]: Oh, that makes more sense. It seemed obvious that the third point is an alternative to the other two. Also all three are options for finding out what ligand is supposed to be used so maybe... 

"Identify your ligand by name or SMILES string from the literature or your collaborators"

**Comment by bannanc** [comment]
@davidlmobley I went ahead and accepted the changes since I understand what your goal was with the last bullet point, I'll leave it up to you if you want to reword.

**Pull request merged**